src/components/SourceEditForm/parser/GridLayout.js

```diff
@@ -0,0 +1,56 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+
+import { Grid } from '@patternfly/react-core/dist/js/layouts/Grid/Grid';
+import { GridItem } from '@patternfly/react-core/dist/js/layouts/Grid/GridItem';
+import { Button } from '@patternfly/react-core/dist/js/components/Button/Button';
+import TrashIcon from '@patternfly/react-icons/dist/js/icons/trash-icon';
+
+import useFormApi from '@data-driven-forms/react-form-renderer/dist/cjs/use-form-api';
+
+import AuthenticationId from './AuthenticationId';
+import sourceEditContext from '../sourceEditContext';
+import RemoveAuthPlaceholder from './RemoveAuthPlaceholder';
+
+const GridLayout = ({ id, fields }) => {
+    const intl = useIntl();
+    const { renderForm } = useFormApi();
+    const { setState, source } = useContext(sourceEditContext);
+
+    const setAuthRemoving = () => setState({
+        type: 'setAuthRemoving',
+        removingAuth: id,
+    });
+
+    const isDeleting = source?.authentications?.find(auth => auth.id === id)?.isDeleting;
+
+    if (isDeleting) {
+        return <RemoveAuthPlaceholder />;
+    }
+
+    return (<Grid>
+        <GridItem md={2} className="ins-c-sources__grid-layout">
```

Comment:
Can you use flex layout instead of grid if you are overriding the display property to flex?